Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+def label = "cognite-grafana-datasource-${UUID.randomUUID().toString().substring(0, 5)}"
+def imageName = "cognite/grafana-cdp"
+def devImageName = "cognite/grafana-cdp-dev"
+
+podTemplate(
+  label: label,
+  containers: [
+    containerTemplate(
+      name: 'node',
+      image: 'node:9',
+      resourceLimitCpu: '2000m',
+      resourceLimitMemory: '1000Mi',
+      ttyEnabled: true
+    ),
+    containerTemplate(
+      name: 'docker',
+      command: '/bin/cat -',
+      image: 'docker:18.06.1-ce',
+      resourceLimitCpu: '1000m',
+      resourceLimitMemory: '1000Mi',
+      ttyEnabled: true
+    ),
+  ],
+  volumes: [
+    secretVolume(secretName: 'npm-credentials',
+                 mountPath: '/npm-credentials',
+                 readOnly: true),
+    secretVolume(secretName: 'cognitecicd-dockerhub', mountPath: '/dockerhub-credentials'),
+    hostPathVolume(hostPath: '/var/run/docker.sock', mountPath: '/var/run/docker.sock'),
+  ],
+  envVars: [
+    envVar(key: 'CHANGE_ID', value: env.CHANGE_ID),
+  ],
+  ) {
+    node(label) {
+      def shortSha
+      container('node') {
+        stage('Checkout') {
+          checkout(scm)
+          shortSha = sh(returnStdout: true, script: 'git rev-parse --short HEAD').trim()
```

Comment:
We've found it quite useful to include the date in the image tag, see https://github.com/cognitedata/transformers-backend/blob/master/Jenkinsfile#L48 for an example.